### ISSUE TITLE
AnalogSGD() lr argument is not considered

### ISSUE THREAD

**Issue by joris-gentinetta** [IBM/aihwkit#449]
## Description
Passing the learning rate to in initialize the AnalogSGD optimizer does not seem to work. However, setting it afterwards with optim.set_learning_rate(LR) works.

## How to reproduce

import torch

from aihwkit.simulator.configs.configs import FloatingPointRPUConfig
from aihwkit.optim import AnalogSGD
from aihwkit.nn import AnalogLinear

rpu_config = FloatingPointRPUConfig()
init_weights = torch.zeros((4,4))

layer = AnalogLinear(4, 4, rpu_config=rpu_config, bias=False, realistic_read_write=False) 
layer.set_weights(init_weights, force_exact=False)

optim = AnalogSGD(layer.parameters(), lr=1) # optim.set_learning_rate(1)

before = layer.analog_tile.get_weights()[0]

x_input = torch.tensor([1, 1, 1, 1], dtype=torch.float32)
d_input = torch.tensor([1, 1, 1, 1], dtype=torch.float32)
layer.analog_tile.update(x_input, d_input)
after = layer.analog_tile.get_weights()[0]
print(after-before)
before = after.numpy().copy()

## Expected behavior

#out (actual):
tensor([[-0.0100, -0.0100, -0.0100, -0.0100],
        [-0.0100, -0.0100, -0.0100, -0.0100],
        [-0.0100, -0.0100, -0.0100, -0.0100],
        [-0.0100, -0.0100, -0.0100, -0.0100]])

#out (expected):
tensor([[-1., -1., -1., -1.],
        [-1., -1., -1., -1.],
        [-1., -1., -1., -1.],
        [-1., -1., -1., -1.]])

## Other information

If you set the learning rate in the following manner, it works: 
optim.set_learning_rate(1)

* Pytorch version: 1.13.0
* Package version: dev install
* OS: macOS Ventura 13.0.1 (Intel)
* Python version: 3.8.15
* Conda version (or N/A): 4.12.0

**Comment by maljoras** [comment]
Hi @joris-gentinetta:
indeed, the learning rate is only set when you actually call `opt.step()`.  If you want to directly manipulate the learning rate on the tile level and call the update separately, then you should use `analog_tile.set_learning_rate()` directly.
